axelrod/strategies/meta.py

```diff
@@ -581,19 +581,29 @@ def __init__(self, p_memory_delete: float = 0.1, p_memory_alter: float = 0.03,
     def __repr__(self):
         return Player.__repr__(self)
 
-    # translates the actions (D and C) to numeric values (loss_value and
-    # gain_value)
-    def gain_loss_tr(self):
-        self.gloss_values = [*map(lambda x: self.loss_value if x == D else
-                                         self.gain_value, self.memory)]
+    def gain_loss_translate(self):
+        """
+        Translates the actions (D and C) to numeric values (loss_value and
+        gain_value).
+        """
+        self.gloss_values = []
+        for action in self.memory:
+            if action == D:
+                self.gloss_values.append(self.loss_value)
+            else:
+                self.gloss_values.append(self.gain_value)
```

Comment:
This is a suggestion rather than a request but Github doesn't really cope with that idea!

```
def gain_loss_translate(self):
    """
    Translates the actions (D and C) to numeric values (loss_value and
    gain_value).
    """
    values = {
        C: self.gain_value,
        D: self.loss_value
    }
    self.gloss_values = [values[action] for action in self.memory]
```

Reply:
I'll go with this, it's more elegant. Thanks!